### PR TITLE
add some instance docs

### PR DESCRIPTION
**Pull request by brainrape**
and improve `Tuple` and `/\` docs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ data Tuple a b
   = Tuple a b
 ```
 
-A simple product type for wrapping pairs of values.
+A simple product type for wrapping a pair of component values.
```
Comment (paf31) [inline README.md]: Why component?

**Comment by paf31** [comment]
Generally, I haven't been documenting instances on purpose. My reasoning is that the only thing that is interesting about an instance is the set of laws it satisfies, and those should be documented with the type class declaration.

**Comment by garyb** [comment]
I don't think it's a problem to document them though - it may not be necessary if you know and understand the implication(s) of the laws, but having descriptions as well may be helpful to a non-Haskell audience.

**Comment by paf31** [comment]
Yeah, I agree with that.

File: README.md (hunk 1 above)
Comment (brainrape) [inline README.md]: The terminology is used throughout the module, and this is where it's introduced.

Comment (paf31) [inline README.md]: I see, thanks. One thing then, we might want to change the docs for `zip`, since "components" should probably be "elements" there.

**Comment by brainrape** [comment]
Instance docs are useful. I would have written one for `bindTuple` (and the rest) if I knew what the idea was behind it.

File: README.md (hunk 1 above)
Comment (brainrape) [inline README.md]: indeed, fixed

**Comment by paf31** [comment]
:+1: Thanks!

**Pull request merged**